### PR TITLE
Feature/fake clock

### PR DESCRIPTION
**Pull request by mkubala**

### CODE REVIEW

File: src/test/scala/com/softwaremill/mqperf/util/FakeClock.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.softwaremill.mqperf.util
+
+import java.util.concurrent.atomic.AtomicLong
+
+import scala.compat.java8.FunctionConverters._
```
Comment (kciesielski) [inline src/test/scala/com/softwaremill/mqperf/util/FakeClock.scala:5]: ❤️

File: src/test/scala/com/softwaremill/mqperf/ReceiverRunnableTest.scala [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    sendMsgAndWaitForReceive()
+
+    receiverShouldBeRunning
+    // Almost max idle time
+    addClockTicks(receiverTimeout - 1.nano)
+    receiverShouldBeRunning
+
+    // Exceed the idle timeout
+    addClockTicks(1.nano)
+    receiverShouldBeStopped
```
Comment (kciesielski) [inline src/test/scala/com/softwaremill/mqperf/ReceiverRunnableTest.scala:56]: I really like the way I can read these tests :+1:

File: src/test/scala/com/softwaremill/mqperf/ReceiverRunnableTest.scala [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  }
+
+  it should "not timeout before receiving any message" in new ReceiverTestCase {
```
Comment (kciesielski) [inline src/test/scala/com/softwaremill/mqperf/ReceiverRunnableTest.scala:59]: I didn't know this `in new X` thing :+1:

Comment (adamw) [reply src/test/scala/com/softwaremill/mqperf/ReceiverRunnableTest.scala:59]: I didn't either, but maybe it's "just" that after `in` you can have an arbitrary code block, here it's new object construction with something done in the constructor?

Comment (kciesielski) [reply src/test/scala/com/softwaremill/mqperf/ReceiverRunnableTest.scala:59]: Looks like it. Good to know though that ScalaTest can resolve constructor's code as failing/successful test (or can it? ;)

Comment (adamw) [reply src/test/scala/com/softwaremill/mqperf/ReceiverRunnableTest.scala:59]: if an exception is thrown ... it doesn't make any difference if it's a constructor or not, right?

Comment (mkubala) [reply src/test/scala/com/softwaremill/mqperf/ReceiverRunnableTest.scala:59]: @adamw exactly. 

It's an alternative to using some `def` that expects a function of N-arguments to be passed or juggling with many helper methods that create mocks or aggregate common assertions (using anonymous trait instances we gain pretty good isolation and there is no need to remember about test initialization).

Comment (kciesielski) [reply src/test/scala/com/softwaremill/mqperf/ReceiverRunnableTest.scala:59]: Well, if a "failing test" is always driven by exception then yes. Which means that a failing test is always a result of an exception, right? (I can't think of any other way if we don't care about return types).

Comment (kciesielski) [reply src/test/scala/com/softwaremill/mqperf/ReceiverRunnableTest.scala:59]: Ok, checked that. It's always exceptions, thx :)

**Pull request merged**